src/main/java/org/apache/commons/crypto/stream/CipherInputStream.java

```diff
@@ -48,7 +51,9 @@
   protected final Cipher cipher;
   protected final int bufferSize;
 
-  protected final byte[] key;
+  protected final Key key;
+  protected final AlgorithmParameterSpec params;
+
   protected final byte[] initIV;
```

Comment:
@jerrychenhf,  exactly!  iv/initIV should be moved to CTRCipherInputStream/CTRCipherOutputStream.  fixed.